src/HistoryQuery/HistoryQueryEngine.class.js

```diff
@@ -55,15 +75,103 @@ class Engine extends BaseEngine {
    * @return {void}
    */
   async start(safeMode = false) {
-    this.logger.warn(`start() should be surcharged ${safeMode}`)
+    this.logger.info('Starting HistoryQueryEngine')
+    if (safeMode || this.check) {
+      this.logger.warn('HistoryQuery is not executed in safe mode')
+      return
+    }
+    this.runNextHistoryQuery()
   }
 
   /**
    * Gracefully stop every Timer, Protocol and Application
    * @return {Promise<void>} - The stop promise
    */
   async stop() {
-    this.logger.warn('stop() should be surcharged')
+    if (this.historyQuery) {
+      await this.historyQuery.stop()
+    }
+  }
+
+  /**
+   * Get the next history query config
+   * @returns {object|null} - The next history query config
+   */
+  getNextHistoryQueryConfig() {
+    const historyQueryConfigs = this.configService.getActiveHistoryQueryConfiguration()
+    const activeHistoryQueryConfigs = historyQueryConfigs.filter((historyQueryConfig) => historyQueryConfig.enabled && !historyQueryConfig.paused)
+
+    const exportingHistoryQueries = activeHistoryQueryConfigs.filter(
```

Comment:
This variable should be renamed since it includes the "exporting" and "importing" status
I suggest ongoingHistoryQueries for example